src/charm.py

```diff
@@ -514,6 +514,11 @@ def _common_exit_hook(self) -> None:
     def _on_pebble_ready(self, _):
         """Event handler for PebbleReadyEvent."""
         self._common_exit_hook()
+        try:
+            version = self.api.version
+            self.unit.set_workload_version(version)
```

Comment:
I'd maybe add this into update_status too in case it fails here?

Reply:
+1 
We have no guarantee that when `pebble_ready` is fired, Alertmanager is up and running....
I would also add this to `upgrade_charm`, since there is a good chance that this led to an workload image upgrade.

Reply:
I agree with adding it to `update_status`

If the charm is upgraded and we get a new container, then `pebble_ready` should fire anyway, so we shouldn't need to do it in `upgrade_charm`.

Reply:
Consider moving `set_workload_version` outside the try-except.

Reply:
@sed-i I had it in an else at first but this seemed more clear to me upon further consideration. It is identical functionally.

Reply:
I'd add it to `upgrade_charm` as well to make the intent clear. It doesn't hurt, but also makes it a bit more reliable if Juju ever makes changes to the lifecycle.

Reply:
I have changed the code to exec `alertmanager --version`. Thus we are not relying on the software actually being ready. So I think it is better not to rely on `update_status`.

I don't think running it in `upgrade_charm` makes any sense. If the lifecycle changes to the point where deploying a new image does not trigger pebble ready, then all of our charms will have much bigger problems than this.